File: platform/extension-sdk/go/platformsdk/telemetry.go
package platformsdk

import "encoding/json"

type TelemetryClient struct {
	Enabled     bool
	ServiceName string
}

func (client TelemetryClient) TrackEvent(name string, properties map[string]string) string {
	if !client.Enabled {
		return ""
	}

	payload, _ := json.Marshal(map[string]any{
		"type":        "platform.telemetry.event",
		"serviceName": client.ServiceName,
		"name":        name,
		"properties":  properties,
	})
	return string(payload)
}

File: platform/extension-sdk/python/src/aks_platform_sdk/telemetry.py
import json
import os
from dataclasses import dataclass


@dataclass(frozen=True)
class PlatformTelemetryClient:
    enabled: bool
    service_name: str

    def track_event(self, name: str, properties: dict[str, str] | None = None) -> None:
        if not self.enabled:
            return

        print(json.dumps({
            "type": "platform.telemetry.event",
            "serviceName": self.service_name,
            "name": name,
            "properties": properties or {},
        }))


def configure_platform_telemetry(
    service_name: str,
    connection_string: str | None = None,
    enabled: bool = True,
) -> PlatformTelemetryClient:
    if not enabled or not connection_string:
        return PlatformTelemetryClient(enabled=False, service_name=service_name)

    os.environ["APPLICATIONINSIGHTS_CONNECTION_STRING"] = connection_string
    os.environ["OTEL_SERVICE_NAME"] = service_name

    from azure.monitor.opentelemetry import configure_azure_monitor

    configure_azure_monitor(connection_string=connection_string)
    return PlatformTelemetryClient(enabled=True, service_name=service_name)
